store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/ParquetStore.java

```diff
@@ -0,0 +1,335 @@
+                    .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+                    .config("spark.kryo.registrator", "uk.gov.gchq.gaffer.spark.serialisation.kryo.Registrator")
+                    .getOrCreate();
+            return super.createContext(new SparkUser(user, spark));
+        }
+    }
+
+    protected void addAdditionalOperationHandlers() {
+        addOperationHandler(GetDataFrameOfElements.class, new GetDataframeOfElementsHandler());
+        addOperationHandler(ImportRDDOfElements.class, new AddElementsFromRDDHandler());
+    }
+
+    @Override
+    protected OutputOperationHandler<GetElements, CloseableIterable<? extends Element>> getGetElementsHandler() {
+        return new GetElementsHandler();
+    }
+
+    @Override
+    protected GetAllElementsHandler getGetAllElementsHandler() {
+        return new GetAllElementsHandler();
+    }
+
+    @Override
+    protected DummyGetAdjacentEntitySeedsHandler getAdjacentIdsHandler() {
+        return new DummyGetAdjacentEntitySeedsHandler();
+    }
+
+    protected OperationHandler<? extends AddElements> getAddElementsHandler() {
+        return new AddElementsHandler();
+    }
+
+    @Override
+    protected Class<? extends Serialiser> getRequiredParentSerialiserClass() {
+        return Serialiser.class;
+    }
+
+    @Override
+    protected Object doUnhandledOperation(final Operation operation, final Context context) {
+        throw new UnsupportedOperationException("Operation " + operation.getClass() + " is not supported by the ParquetStore.");
+    }
+
+    @SuppressFBWarnings(value = "BC_UNCONFIRMED_CAST_OF_RETURN_VALUE", justification = "The properties should always be ParquetStoreProperties")
+    @Override
+    public ParquetStoreProperties getProperties() {
+        return (ParquetStoreProperties) super.getProperties();
+    }
+
+    public void loadIndices() throws StoreException {
+        try {
+            if (this.groupToIndex == null) {
+                this.groupToIndex = new HashMap<>();
+            } else {
+                this.groupToIndex.clear();
+            }
+            final FileSystem fs = getFS();
+            final String rootDirString = this.getProperties().getDataDir();
+            final Path rootDir = new Path(rootDirString);
+            if (fs.exists(rootDir)) {
+                final FileStatus[] snapshots = fs.listStatus(rootDir);
+                long latestSnapshot = getCurrentSnapshot();
+                for (final FileStatus fileStatus : snapshots) {
+                    final String snapshot = fileStatus.getPath().getName();
+                    long currentSnapshot = Long.parseLong(snapshot);
+                    if (latestSnapshot < currentSnapshot) {
+                        latestSnapshot = currentSnapshot;
+                    }
+                }
+                if (latestSnapshot != 0L) {
+                    setCurrentSnapshot(latestSnapshot);
```

Comment:
I'm concerned that this latestSnapshot variable could be modified in a different thread whilst you are using it here.